motivaa-backend/motivaa-app/src/main/java/com/motivaa/control/HabitFinderService.java

```diff
@@ -0,0 +1,33 @@
+package com.motivaa.control;
+
+import com.motivaa.control.error_handling.exceptions.RepositoryException;
+import com.motivaa.control.repository.MotivaaRepository;
+import com.motivaa.control.utility.MessageBundle;
+import com.motivaa.entity.Habit;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.UUID;
+
+@Log4j2
+@Service
+public class HabitFinderService {
+    MotivaaRepository motivaaRepository;
+
+    public HabitFinderService(MotivaaRepository motivaaRepository) {
+        this.motivaaRepository = motivaaRepository;
+    }
+
+    public List<Habit> habitFinder(String userUuid) {
+        List<Habit> habits;
+        try {
+            habits = motivaaRepository.searchHabitByUserUuid(UUID.fromString(userUuid));

```

Comment:
I don't necessarily think it is a problem but worth to think through:
You will search for habits with uuid even if no user is present in the system with the given uuid. 

In normal curciumstanses this would not cause any problem, but for example if you will be required to delete a user, you will have to make sure that you delete the habits as well, otherwise you could still return it with this api.
Are there any other scenarios where this could be a problem?

I wouldn't check if the user exists only if it is necessary because it is extra operation toward elasticsearch. So jut think through if it is okay like this or not.

Reply:
I also occured to me that since you don't check if the uuid belongs to an actual user, you would give back empty list even though the uuid does not belong to any user. Could this be a problem?

Reply:
The empty list if there is no such user was a conscious choice: I found it a 50/50 choice and I liked this route better mainly because the other would double the amount of db interactions.

The counterargument that comes to my head is that now returning and empty list could mean two things and that could be ambiguous. I think this doesnt outweigh the double search.

Thanks for raising - it was better to think it through twice then none!